### PR TITLE
Make biopython an optional dependency

### LINKED ISSUE #3820: Make biopython an optional dependency
I think this has been a long time request by @jbarnoud (apologies if I'm misremembering).

Not the biggest dependency, but we could easily make it an optional one given we need it mostly for sequence alignment. Probably needs some discussion re: the impact it has on seqres though.

### PR DESCRIPTION
**Pull request by IAlibay**
Fixes #3820 

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

## Developers certificate of origin
- [x] I certify that this contribution is covered by the LGPLv2.1+ license as defined in our [LICENSE](https://github.com/MDAnalysis/mdanalysis/blob/develop/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

----
:books: Documentation preview :books:: https://mdanalysis--4332.org.readthedocs.build/en/4332/

### CODE REVIEW

File: package/MDAnalysis/analysis/align.py [hunk 1]
```diff
@@ -1057,7 +1066,16 @@ def sequence_alignment(mobile, reference, match_score=2, mismatch_penalty=-1,
        Replace use of deprecated :func:`Bio.pairwise2.align.globalms` with
        :class:`Bio.Align.PairwiseAligner`.
 
+    .. versionchanged:: 2.7.0
+       Biopython is now an optional dependency which this method requires.
+
     """
+    if not HAS_BIOPYTHON:
+        errmsg = ("The `sequence_alignment` method requires an installation "
+                  "of biopython. Please install biopython to use this method: "
```
Comment (RMeli) [inline package/MDAnalysis/analysis/align.py]:
```suggestion
                  "of Biopython. Please install Biopython to use this method: "
```
?

File: package/MDAnalysis/analysis/align.py [hunk 2]
```diff
@@ -1159,15 +1177,29 @@ def fasta2select(fastafilename, is_aligned=False,
...
     .. versionchanged:: 1.0.0
        Passing `alnfilename` or `treefilename` as `None` will create a file in
        the current working directory.
+    .. versionchanged:: 2.7.0
+       Biopython is now an optional dependency which this method requires.
 
     """
+    if not HAS_BIOPYTHON:
+        errmsg = ("The `fasta2select` method requires an installation "
+                  "of biopython. Please install biopython to use this method: "
```
Comment (RMeli) [inline package/MDAnalysis/analysis/align.py]:
```suggestion
                  "of Biopython. Please install Biopython to use this method: "
```

Maybe this message could be written once as module variable? `missing_biopython_errmsg` or something on those lines?

File: package/MDAnalysis/core/topologyattrs.py [hunk 3]
```diff
@@ -2810,9 +2816,20 @@ def sequence(self, **kwargs):
...
+
 
         .. versionadded:: 0.9.0
+        .. versionchanged:: 2.7.0
+          Biopython is now an optional dependency
         """
+        if not HAS_BIOPYTHON:
+            errmsg = ("The `sequence_alignment` method requires an "
+                      "installation of biopython. Please install "
+                      "biopython to use this method: "
```
Comment (RMeli) [inline package/MDAnalysis/core/topologyattrs.py]:
```suggestion
                      "installation of Biopython. Please install "
                      "Biopython to use this method: "
```
?

File: package/MDAnalysis/analysis/align.py [hunk 4]
```diff
@@ -196,10 +196,14 @@
...
 import numpy as np
 
-import Bio.SeqIO
-import Bio.AlignIO
-import Bio.Align
-import Bio.Align.Applications
+try:
+    import Bio.AlignIO
+    import Bio.Align
+    import Bio.Align.Applications
```
Comment (RMeli) [inline package/MDAnalysis/analysis/align.py:202]: Mmmh, why is this not covered?

Comment (IAlibay) [reply package/MDAnalysis/analysis/align.py:202]: Not sure - I added a quick test in [5529a5d](https://github.com/MDAnalysis/mdanalysis/pull/4332/commits/5529a5d7ce7f627f9e58af80d3374d735a637638) to demonstrate that the test was being called, and it is. This, like the networkx issue, seems to be a codecov thing?

Comment (RMeli) [reply package/MDAnalysis/analysis/align.py:202]: Ok, thanks for checking. I thought it might be the same issue you outlined with NetworkX

Comment (IAlibay) [reply package/MDAnalysis/analysis/align.py:202]: Can confirm at least two hits on codecov: https://app.codecov.io/gh/MDAnalysis/mdanalysis/pull/4332?src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=checks&utm_campaign=pr+comments&utm_term=MDAnalysis

**Review by RMeli — CHANGES_REQUESTED** [review]
I think this needs a changelog entry?

File: package/MDAnalysis/analysis/align.py (hunk 2 above)
Comment (IAlibay) [reply package/MDAnalysis/analysis/align.py]: missed that comment - could do but we're using slightly different method names. I don't have any strong feelings here so I'm happy to change things up if you think it's worth it.

Comment (RMeli) [reply package/MDAnalysis/analysis/align.py]: Oh, right. I missed the different method name. Never mind.

Comment (RMeli) [reply package/MDAnalysis/analysis/align.py]: Might be a good use of string templates, but I don't think it's worth it.

File: package/MDAnalysis/core/topologyattrs.py [hunk 5]
```diff
@@ -2810,9 +2816,20 @@ def sequence(self, **kwargs):
 
         :exc:`TypeError` if an unknown *format* is selected.
 
+        :exc:`ImportError` is the biopython package is not available.
```
Comment (RMeli) [inline package/MDAnalysis/core/topologyattrs.py]:
```suggestion
        :exc:`ImportError` is the Biopython package is not available.
```

File: package/MDAnalysis/core/topologyattrs.py [hunk 6]
```diff
@@ -2810,9 +2816,20 @@ def sequence(self, **kwargs):
...
+
 
         .. versionadded:: 0.9.0
+        .. versionchanged:: 2.7.0
+          Biopython is now an optional dependency
```
Comment (RMeli) [inline package/MDAnalysis/core/topologyattrs.py]:
```suggestion
        .. versionchanged:: 2.7.0
           Biopython is now an optional dependency
```

**Review by RMeli — APPROVED** [review]
Last nitpicks but LGTM

**Pull request merged**